tests/model_indexes/tests.py

```diff
@@ -100,14 +114,24 @@ class Meta:
         self.assertEqual(index.name, 't_quoted_name_e4ed1b_idx')
 
     def test_deconstruction(self):
-        index = models.Index(fields=['title'], db_tablespace='idx_tbls')
+        index = models.Index(
+            name='model_index_title_196f42_idx',
+            fields=['title'],
+            db_tablespace='idx_tbls',
+            condition=Q(pages__gt=400),
+        )
         index.set_name_with_model(Book)
         path, args, kwargs = index.deconstruct()
         self.assertEqual(path, 'django.db.models.Index')
         self.assertEqual(args, ())
         self.assertEqual(
             kwargs,
-            {'fields': ['title'], 'name': 'model_index_title_196f42_idx', 'db_tablespace': 'idx_tbls'}
+            {
+                'fields': ['title'],
+                'name': 'model_index_title_196f42_idx',
+                'db_tablespace': 'idx_tbls',
+                'condition': Q(pages__gt=400),
+            }
         )
```

Comment:
Perhaps we should continue to test the simpler case where we don't provide a `name` or `condition`.

Reply:
But then some branches are not tested.

Reply:
I meant add these changes as a new test, instead of modifying the existing one.

Reply:
A new test has been added.